### ISSUE TITLE
Add Timebank Functionality for Handling Extra Hours Worked

### ISSUE THREAD

**Issue by olemp** [Puzzlepart/did#1084]
- [x] I have searched for existing issues to avoid creating duplicates

### TLDR
We propose adding a "timebank" feature to did that allows employees to request time off after working extra hours, and for HR and management to view and approve these requests and track timebank balances. This will provide a more organized and efficient way to manage extra hours worked and time off.

### Description
As a company, we want to provide our employees with the ability to track and manage the extra hours they work. This is important for ensuring that employees are fairly compensated for their extra time and effort, and for managing workload and staffing levels within the company.

To address this, we propose the addition of a "timebank" feature to did. This feature would allow employees to request to have extra hours worked added to their timebank balance. The timebank balance could then be used to request time off at a later date, either as paid time off or unpaid time off as determined by company policy.

The timebank feature would also provide HR and managers with the ability to view and approve requests to add extra hours worked to an employee's timebank balance, as well as view the timebank balances of individual employees.

Overall, the timebank feature would provide a streamlined and organized way for our employees to track and request time off for extra hours worked, while also giving HR and management visibility into the extra hours worked by the company's employees.

### Proposed solution
This could be done by tagging events somehow.

<img width="555" alt="image" src="https://user-images.githubusercontent.com/7606007/208897113-0fe77195-7421-4bf5-a11f-b2b67e2f4941.png">

<img width="422" alt="image" src="https://user-images.githubusercontent.com/7606007/208899949-8b070980-0d30-434d-b9b0-28aa6db9752f.png">
